arcgis-ios-sdk-samples/Maps/Generate offline map (overrides)/OfflineMapParameterOverridesViewController.swift

```diff
@@ -0,0 +1,263 @@
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import UIKit
+import ArcGIS
+
+class OfflineMapParameterOverridesViewController: UITableViewController {
+    
+    var parameterOverrides: AGSGenerateOfflineMapParameterOverrides?
+    var map: AGSMap?
+    
+    //MARK: - Outlets
+
+    /// The min scale level for the output. Note that lower values are zoomed further out,
+    /// i.e. 0 has the least detail, but one tile covers the entire Earth.
+    @IBOutlet weak var minScaleLevelSlider: UISlider!
+    
+    /// The max scale level for the output. Note that higher values are zoomed further in,
+    /// i.e. 23 has the most detail, but each tile covers a tiny area.
+    @IBOutlet weak var maxScaleLevelSlider: UISlider!
+    
+    /// The extra padding added to the extent rect to fetch a larger area, in meters.
+    @IBOutlet weak var basemapExtentBufferSlider: UISlider!
+    
+    /// Switch indicating if the system valves layer should be included in the download.
+    @IBOutlet weak var includeSystemValvesSwitch: UISwitch!
+    ///Switch indicating if the service connections layer should be included in the download.
+    @IBOutlet weak var includeServiceConnectionsSwitch: UISwitch!
+    
+    /// The minimum flow rate by which to filter features in the Hydrants layer, in gallons per minute.
+    @IBOutlet weak var minHydrantFlowRateSlider: UISlider!
+    
+    ///Switch indicating if the pipe layers should be restricted to the extent frame.
+    @IBOutlet weak var cropWaterPipesToExtentSwitch: UISwitch!
+    
+    //MARK: - Actions
+    
+    @IBAction func sliderChangeAction(_ sender: UISlider) {
+        if sender == minScaleLevelSlider {
+            // Disallow a min value greater than the maximum
+            if sender.value > maxScaleLevelSlider.value {
+                sender.value = maxScaleLevelSlider.value
+            }
+        }
+        else if sender == maxScaleLevelSlider {
+            // Disallow a max value less than the minimum
+            if sender.value < minScaleLevelSlider.value {
+                sender.value = minScaleLevelSlider.value
+            }
+        }
+        // Update this slider's text field to display the new value
+        updateTextField(for: sender)
+    }
+    
+    //MARK: - Text field updating
+    
+    private func updateTextField(for slider: UISlider){
+        let numberFormatter = NumberFormatter()
+        if let text = numberFormatter.string(from: slider.value as NSNumber),
+            let superview = slider.superview,
+            let field = superview.subviews.first(where: { $0 is UITextField }) as? UITextField{
```

Comment:
I designed it this way so all the sliders can all use the same action. Otherwise we'd need to know what slider action corresponds to what text field, so we'd either need two more actions or three slider outlets as well as three text field outlets. It also scales to adding more slider cells in the future, all without subclassing the table view cells.

I agree it's somewhat less efficient but not noticeably so.

Reply:
You would need more outlets and/or more action methods, but I think it would be worth it. I was confused the first time I saw this method. And when I did figure out what was going on, I was even more confused. The more I think about it, the more and more it looks like a hack. Adding outlets for the value labels and action methods for each slider takes a little more time, but it saves a subview lookup *every single time the slider value changes*. Is the lookup a big deal in this case? No. Is it a bad practice in general? Yes.

I'm not opposed to using hacks, but I am opposed to using them if there's another way. That being said, the sample does work and the code in question isn't demonstrating any of the `ArcGIS` API, so I'll approve it the way it is.

Reply:
I changed this to use outlets since I had to differentiate the labels anyway when I changed them to also use UILabel. I had forgotten that I already had slider outlets so it wasn't as much added code as I had imagined.